activity/views.py

```diff
@@ -165,6 +170,27 @@ class MultipleUserNameError(Exception):
     """Existing Username Error"""
     pass
 
+# https://python-social-auth.readthedocs.io/en/latest/pipeline.html#authentication-pipeline
+# ? Are these signatures the same for Microsoft?
+def remove_inactive_user(uid, **kwargs):
+    """
+    If there exists an INACTIVE user with the same email, delete it and its ActivityUser
+    """
+    if User.objects.filter(email=uid, is_active=False).exists():
+        existing_user = User.objects.get(email=uid, is_active=False)
+        existing_user.activity_user.delete()
+        existing_user.delete()
+
+# https://python-social-auth.readthedocs.io/en/latest/pipeline.html#authentication-pipeline
+def handle_new_social_login(response, user, is_new, details, username, new_association, social, **kwargs):
+    """
+    Create an ActivityUser for social logins with unused emails
+    """
+    if new_association and not ActivityUser.objects.filter(user=user).exists():
```

Comment:
@Kimaiyo077 This is the function responsible for creating an ActivityUser for a new social account

Reply:
@Kimaiyo077 Is this happening to you in your local setup? Are you trying different Google accounts? And is the normal signup/login working fine?

Reply:
@TAnas0 Yes it is happening in my local setup. The normal registration and login are working fine. I have tried different google accounts, some are from verified users and others from new users but I am getting the same error.